CMakeLists.txt

```diff
@@ -30,6 +30,11 @@ function(add_plugin target)
     configure_plugin(${target})
 endfunction()
 
+function(add_shared_plugin target)
+    add_library(${target} SHARED ${ARGN})
+    configure_plugin(${target})
+endfunction()
+
```

Comment:
PInvoke cannot accessed exported symbols of "Module" visible libraries, so I added a new function here with different visibility.

I can inline this directly in `DotNET/CMakeLists` if you do not want this added here.